### PR TITLE
ncm-openldap: Allow the checkpoint option to be set

### PR DESCRIPTION
**Pull request by itkovian**
This options allows setting a checkpoint in LDAP's Berkeley DB transaction logs.

### CODE REVIEW

File: ncm-openldap/src/main/perl/openldap.pm [hunk 1]
```diff
@@ -205,6 +205,12 @@ sub print_database_class
         delete($tree->{suffix});
     }
 
+    if (exists($tree->{checkpoint})) {
+        $s = join(" ", $tree->{checkpoint});
```
Comment (jrha) [inline ncm-openldap/src/main/perl/openldap.pm]: How about `my $s`?

File: ncm-openldap/src/main/perl/openldap.pm [hunk 2]
```diff
@@ -449,6 +455,11 @@ sub legacy_setup
         print $fh "suffix $suffix\n";
     }
 
+    if ($config->elementExists("$base/checkpoint")) {
+        $checkpoint = join(" ", $config->getValue($base/checkpoint""));
```
Comment (jrha) [inline ncm-openldap/src/main/perl/openldap.pm]: Also here, `my $checkpoint`

File: ncm-openldap/src/main/perl/openldap.pm [hunk 3]
```diff
@@ -449,6 +455,11 @@ sub legacy_setup
         print $fh "suffix $suffix\n";
     }
 
+    if ($config->elementExists("$base/checkpoint")) {
+        my $checkpoint = join(" ", $config->getValue($base/checkpoint""));
```
Comment (jrha) [inline ncm-openldap/src/main/perl/openldap.pm]: Mangled string? Looks like it should be `"$base/checkpoint"` ?

**Comment by stdweird** [comment]
@itkovian can you add `ncm-openldap` to teh beginning of the title of the PR
also, there weren't any schema changes required to get this running?

**Comment by jrha** [comment]
@itkovian I've changed the title for you, does this not require change to the `ldap_database` type in the schema?

File: ncm-openldap/src/main/perl/openldap.pm [hunk 4]
```diff
@@ -449,6 +455,11 @@ sub legacy_setup
         print $fh "suffix $suffix\n";
     }
 
+    if ($config->elementExists("$base/checkpoint")) {
+        my $checkpoint = join(" ", $config->getValue("$base/checkpoint"));
```
Comment (stdweird) [inline ncm-openldap/src/main/perl/openldap.pm]: does this even work? join on array ref?

File: ncm-openldap/src/main/perl/openldap.pm [hunk 5]
```diff
@@ -205,6 +205,12 @@ sub print_database_class
         delete($tree->{suffix});
     }
 
+    if (exists($tree->{checkpoint})) {
+        my $s = join(" ", $tree->{checkpoint});
```
Comment (stdweird) [inline ncm-openldap/src/main/perl/openldap.pm]: does this even work?

**Comment by jrha** [comment]
@itkovian any update?

**Comment by stdweird** [comment]
should be covered by #476

**Comment by jrha** [comment]
Replaced by #476.

**Pull request closed without merging**